src/lakefs_spec/spec.py

```diff
@@ -116,36 +118,88 @@ def __init__(
 
     @property
     def transaction(self):
-        """A context within which files are committed together upon exit
+        """
+        A context manager within which file uploads and versioning operations are deferred to a
+        queue, and carried during __exit__.
 
-        Requires the file class to implement `.commit()` and `.discard()`
-        for the normal and exception cases.
+        Requires the file class to implement `.commit()` and `.discard()` for the normal and exception cases.
         """
         self._transaction: LakeFSTransaction | None
         if self._transaction is None:
             self._transaction = LakeFSTransaction(self)
         return self._transaction
 
     def start_transaction(self):
-        """Begin write transaction for deferring files, non-context version"""
+        """
+        Prepare a lakeFS file system transaction without entering the transaction context yet.
+        """
         self._intrans = True
         self._transaction = LakeFSTransaction(self)
         return self.transaction
 
     @contextmanager
     def wrapped_api_call(self, message: str | None = None, set_cause: bool = True) -> EmptyYield:
+        """
+        A context manager to wrap lakeFS API calls, translating API errors to Python-native OS errors if they occur.
```

Comment:
```suggestion
        Context manager to wrap lakeFS API calls, translating API errors to Python-native OS errors.
```

Reply:
I would in general prefer omit the articles. But I'm not emotional attached to whatever we agree upon.